monochart/templates/hpa.yaml

```diff
@@ -0,0 +1,44 @@
+{{- $root := . -}}
+{{- if .Values.hpa -}}
+{{- if .Values.hpa.enabled -}}
+apiVersion: autoscaling/v2beta2
+kind: HorizontalPodAutoscaler
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: {{ include "common.fullname" .  }}
+{{- with .Values.deployment.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  labels:
+{{- include "common.labels.standard" . | nindent 4 }}
+{{- with .Values.deployment.labels }}
+{{ toYaml .| indent 4 }}
+{{- end }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "common.fullname" .  }}
+  minReplicas: {{ .Values.hpa.autoscaling.minReplicas }}
+  maxReplicas: {{ .Values.hpa.autoscaling.maxReplicas }}
+  metrics:
+  {{- if .Values.hpa.autoscaling.targetCPUUtilizationPercentage }}
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: {{ .Values.hpa.autoscaling.targetCPUUtilizationPercentage }}
+  {{- end }}
+  {{- if .Values.hpa.autoscaling.targetMemoryUtilizationPercentage }}
+  - type: Resource
+    resource:
+      name: memory
+      target:
+        type: Utilization
+        averageUtilization: {{ .Values.hpa.autoscaling.targetMemoryUtilizationPercentage }}
+  {{- end }}
```

Comment:
This is great, provides a simple interface for the common two use cases. Perhaps we should have an additional section that is more generic until we find other use cases, something like:

```
  {{- if .Values.hpa.autoscaling.metrics }}
{{- include "Values.hpa.autoscaling.metrics" | nindent 2 }}
  {{- end }}
```

(very much pseudocode, not tested)

Reply:
We can add this suggestion to the next iteration if thats ok? That way we can test with the suggested deployment in thread https://spoton.slack.com/archives/CFBPX8NMQ/p1724065763213799